nodes/usb_cam_node.cpp

```diff
@@ -334,12 +343,123 @@ class UsbCamNode
       }
     }
 
-    // publish the image
-    image_pub_.publish(*img_ptr, *ci);
+    if (rotate_code_ == ROTATE_90_CW || rotate_code_ == ROTATE_90_CCW || rotate_code_ == ROTATE_180)
```

Comment:
It would be better to make the header update logic a separate function and add a simple test.